File: tp1/TP01Q07 - Is em C/metodos.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <locale.h>
#include<stdbool.h>
#define TAM 1000 //definição de TAM para 1000 para facilitar ajustes no código

bool vogais(char *linha, int tamanho){
	bool resp = true;
	for(int i = 0; i < tamanho; i++){
		if(linha[i] != 'a' || linha[i] != 'e' || linha[i] != 'i' || linha[i] != 'o' || linha[i] != 'u' || linha[i] != 'A' || linha[i] != 
		'E' || linha[i] != 'I' || linha[i] != 'O' || linha[i] != 'U'){
			resp = false;
			return resp;
		}
	}
	return resp;
}

bool consoantes(char *linha, int tamanho){
	bool resp = true;
	for(int i = 0; i < tamanho; i++){
		if(linha[i] != 'b' || linha[i] != 'c' || linha[i] != 'd' || linha[i] != 'f' || linha[i] != 'g' || linha[i] != 'h' || linha[i] != 
		'j' || linha[i] != 'k' || linha[i] != 'l' || linha[i] != 'm' || linha[i] != 'n' || linha[i] != 'p' || linha[i] != 'q' || 
		linha[i] != 'r' || linha[i] != 's' || linha[i] != 't' || linha[i] != 'v' || linha[i] != 'w' || linha[i] != 'x' || linha[i] != 'y' 
		|| linha[i] != 'z' || linha[i] != 'B' || linha[i] != 'C' || linha[i] != 'D' || linha[i] != 'F' || linha[i] != 
		'G' || linha[i] != 'H' || linha[i] != 'J' || linha[i] != 'K' || linha[i] != 'L' || linha[i] != 'M' || linha[i] != 'N' || 
		linha[i] != 'P' || linha[i] != 'Q' || linha[i] != 'R' || linha[i] != 'S' || linha[i] != 'T' || linha[i] != 'V' || linha[i] != 'W' 
		|| linha[i] != 'X' || linha[i] != 'Y' || linha[i] != 'Z'){
			resp = false;
			return resp;
		}
	}
	return resp;
}

bool numeros_inteiros(char *linha, int tamanho){
	bool resp = true;
	for(int i = 0; i < tamanho; i++){
		if(linha[i] == 'a' || linha[i] == 'e' || linha[i] == 'i' || linha[i] == 'o' || linha[i] == 'u' || linha[i] == 'A' || linha[i] == 
		'E' || linha[i] == 'I' || linha[i] == 'O' || linha[i] == 'U' || linha[i] == 'b' || linha[i] == 'c' || linha[i] == 'd' || linha[i] 
		== 'f' || linha[i] == 'g' || linha[i] == 'h' || linha[i] == 'j' || linha[i] == 'k' || linha[i] == 'l' || linha[i] == 'm' || 
		linha[i] == 'n' || linha[i] == 'p' || linha[i] == 'q' || linha[i] == 'r' || linha[i] == 's' || linha[i] == 't' || linha[i] == 'v' 
		|| linha[i] == 'w' || linha[i] == 'x' || linha[i] == 'y' || linha[i] == 'z' || linha[i] == 'B' || linha[i] == 
		'C' || linha[i] == 'D' || linha[i] == 'F' || linha[i] == 'G' || linha[i] == 'H' || linha[i] == 'J' || linha[i] == 'K' || linha[i] 
		== 'L' || linha[i] == 'M' || linha[i] == 'N' || linha[i] == 'P' || linha[i] == 'Q' || linha[i] == 'R' || linha[i] == 'S' || 
		linha[i] == 'T' || linha[i] == 'V' || linha[i] == 'W' || linha[i] == 'X' || linha[i] == 'Y' || linha[i] == 'Z'  
		|| linha[i] == '.' || linha[i] == ',' || linha[i] == ';' || linha[i] == ':' || linha[i] == '/' || linha[i] == '?' || linha[i] == 
		'!'){
			resp = false;
			return resp;
		}
	}
	return resp;
}

bool numeros_reais(char *linha, int tamanho){
	bool resp = true;
	for(int i = 0; i < tamanho; i++){
		if(linha[i] == 'a' || linha[i] == 'e' || linha[i] == 'i' || linha[i] == 'o' || linha[i] == 'u' || linha[i] == 'A' || linha[i] == 
		'E' || linha[i] == 'I' || linha[i] == 'O' || linha[i] == 'U' || linha[i] == 'b' || linha[i] == 'c' || linha[i] == 'd' || linha[i] 
		== 'f' || linha[i] == 'g' || linha[i] == 'h' || linha[i] == 'j' || linha[i] == 'k' || linha[i] == 'l' || linha[i] == 'm' || 
		linha[i] == 'n' || linha[i] == 'p' || linha[i] == 'q' || linha[i] == 'r' || linha[i] == 's' || linha[i] == 't' || linha[i] == 'v' 
		|| linha[i] == 'w' || linha[i] == 'x' || linha[i] == 'y' || linha[i] == 'z' || linha[i] == 'B' || linha[i] == 
		'C' || linha[i] == 'D' || linha[i] == 'F' || linha[i] == 'G' || linha[i] == 'H' || linha[i] == 'J' || linha[i] == 'K' || linha[i] 
		== 'L' || linha[i] == 'M' || linha[i] == 'N' || linha[i] == 'P' || linha[i] == 'Q' || linha[i] == 'R' || linha[i] == 'S' || 
		linha[i] == 'T' || linha[i] == 'V' || linha[i] == 'W' || linha[i] == 'X' || linha[i] == 'Y' || linha[i] == 'Z'  
		|| linha[i] == ';' || linha[i] == ':' || linha[i] == '/' || linha[i] == '?' || linha[i] == '!'){;
			resp = false;
			return resp;
		}
	}
	return resp;
}

int main(){
	char linha[TAM];
    int i = 1, tamanho = 0;
    bool resp1 = false, resp2 = false, resp3 = false, resp4 = false;
    
    scanf(" %[^\n]", linha); //leitura da primeira linha do pub.in

	if(strcmp(linha, "FIM") == 0){ //comparação do array frase com FIM que corresponde à última linha do arquivo pub.in
        i = 0;
    }

    while(i != 0){	//loop para leitura das linhas do arquivo

 		tamanho = strlen(linha); //tamanho do array
 		
 		resp1 = vogais(linha, tamanho);
 		resp2 = consoantes(linha, tamanho);
 		resp3 = numeros_inteiros(linha, tamanho);
 		resp4 = numeros_reais(linha, tamanho);

        if(resp1 == true){ 
            printf("SIM ");
        }else{
            printf("NAO ");
        }
        
		if(resp2 == true){ 
            printf("SIM ");
        }else{
            printf("NAO ");
        } 
        
        if(resp3 == true){ 
            printf("SIM ");
        }else{
            printf("NAO ");
        }
        
        if(resp4 == true){ 
            printf("SIM \n");
        }else{
            printf("NAO \n");
        }
       
        scanf(" %[^\n]", linha); //leitura de linha a linha do array
        
        if(strcmp(linha, "FIM") == 0){ //condição para ver se a frase lida corresponde ao final do arquivo identificado por FIM
            i = 0;
        }
	}
	return 0;
}

File: tp1/TP01Q06 - Is em Java/metodos.java
import java.io.*;

class metodos{
	public static void main (String[] args){
		String linha = ""; //linha do pub.in
		int i = 0;
		boolean resp1 = false, resp2 = false, resp3 = false, resp4 = false;
	
		while(i != 1){ //loop para leitura de todo arquivo
			linha = MyIO.readLine();
			
			if(linha.charAt(0) == 'F' && linha.charAt(1) == 'I' && linha.charAt(2) == 'M'){  //FIM presente no fim do arquivo
				i = 1; 																		
			}else{
				resp1 = vogais(linha);
				resp2 = consoantes(linha);
				resp3 = numero_inteiro(linha);
				resp4 = numero_real(linha);
				
				if(resp1 == true){
					MyIO.print("SIM ");
				}else{
					MyIO.print("NAO ");
				}
				
				if(resp2 == true){
					MyIO.print("SIM ");
				}else{
					MyIO.print("NAO ");
				}
				
				if(resp3 == true){
					MyIO.print("SIM ");
				}else{
					MyIO.print("NAO ");
				}
				
				if(resp4 == true){
					MyIO.println("SIM");
				}else{
					MyIO.println("NAO");
				}
			}
		}
	}
	
	public static boolean vogais(String linha){ //metodo para conferir se a linha possui apenas vogais
		boolean resp0 = false;
		boolean resp1 = true;
		for(int i = 0; i < linha.length(); i++){	//loop q se repete enquanto i < tamanho da linha			
        	if(linha.charAt(i) != 'a' || linha.charAt(i) != 'e' || linha.charAt(i) != 'i' || linha.charAt(i) != 'o' || linha.charAt(i) != 
        	'u' || linha.charAt(i) != 'A' || linha.charAt(i) != 'E' || linha.charAt(i) != 'I' || linha.charAt(i) != 'O' || 
        	linha.charAt(i) != 'U'){
				return resp0;
			}
        }
        return resp1;
	}
	
	public static boolean consoantes(String linha){ //metodo para conferir se a linha possui apenas consoantes	
		boolean resp0 = false;
		boolean resp1 = true;
		for(int i = 0; i < linha.length(); i++){	//loop q se repete enquanto i < tamanho da linha			
        	if(linha.charAt(i) != 'b' || linha.charAt(i) != 'c' || linha.charAt(i) != 'd' || linha.charAt(i) != 'f' || linha.charAt(i) != 
        	'g' || linha.charAt(i) != 'h' ||linha.charAt(i) != 'j' || linha.charAt(i) != 'k' || linha.charAt(i) != 'l' || 
        	linha.charAt(i) != 'm' || linha.charAt(i) != 'n' || linha.charAt(i) != 'p' || linha.charAt(i) != 'q' || linha.charAt(i) != 
        	'r' || linha.charAt(i) != 's' || linha.charAt(i) != 't' || linha.charAt(i) != 'v' || linha.charAt(i) != 'w' || 		
        	linha.charAt(i) != 'x' || linha.charAt(i) != 'y' || linha.charAt(i) != 'z' || linha.charAt(i) != 'B' || linha.charAt(i) != 
        	'C' || linha.charAt(i) != 'D' || linha.charAt(i) != 'F' || linha.charAt(i) != 'G' || linha.charAt(i) != 'H' || 
        	linha.charAt(i) != 'J' || linha.charAt(i) != 'K' || linha.charAt(i) != 'L' || linha.charAt(i) != 'M' || linha.charAt(i) != 
        	'N' || linha.charAt(i) != 'P' || linha.charAt(i) != 'Q' || linha.charAt(i) != 'R' || linha.charAt(i) != 'S' || 
        	linha.charAt(i) != 'T' || linha.charAt(i) != 'V' || linha.charAt(i) != 'W' || linha.charAt(i) != 'X' || linha.charAt(i) != 
        	'Y' || linha.charAt(i) != 'Z'){
				return resp0;
			} 
        }
        return resp1;
	}
	
	public static boolean numero_inteiro(String linha){ //metodo para conferir se a linha possui apenas números inteiros	
		boolean resp0 = false;
		boolean resp1 = true;
		for(int i = 0; i < linha.length(); i++){	//loop q se repete enquanto i < tamanho da linha			
        	if(linha.charAt(i) != 'a' || linha.charAt(i) != 'e' || linha.charAt(i) != 'i' || linha.charAt(i) != 'o' || linha.charAt(i) != 
        	'u' || linha.charAt(i) != 'A' || linha.charAt(i) != 'E' || linha.charAt(i) != 'I' || linha.charAt(i) != 'O' || 
        	linha.charAt(i) != 'U'|| linha.charAt(i) != 'b' || linha.charAt(i) != 'c' || linha.charAt(i) != 'd' || linha.charAt(i) != 'f' 
        	|| linha.charAt(i) != 'g' || linha.charAt(i) != 'h' ||linha.charAt(i) != 'j' || linha.charAt(i) != 'k' || linha.charAt(i) != 
        	'l' || linha.charAt(i) != 'm' || linha.charAt(i) != 'n' || linha.charAt(i) != 'p' || linha.charAt(i) != 'q' || 
        	linha.charAt(i) != 'r' || linha.charAt(i) != 's' || linha.charAt(i) != 't' || linha.charAt(i) != 'v' || linha.charAt(i) != 
        	'w' || linha.charAt(i) != 'x' || linha.charAt(i) != 'y' || linha.charAt(i) != 'z' || linha.charAt(i) != 'B' || 
        	linha.charAt(i) != 'C' || linha.charAt(i) != 'D' || linha.charAt(i) != 'F' || linha.charAt(i) != 'G' || linha.charAt(i) != 
        	'H' || linha.charAt(i) != 'J' || linha.charAt(i) != 'K' || linha.charAt(i) != 'L' || linha.charAt(i) != 'M' || 
        	linha.charAt(i) != 'N' || linha.charAt(i) != 'P' || linha.charAt(i) != 'Q' || linha.charAt(i) != 'R' || linha.charAt(i) != 
        	'S' || linha.charAt(i) != 'T' || linha.charAt(i) != 'V' || linha.charAt(i) != 'W' || linha.charAt(i) != 'X' || 
        	linha.charAt(i) != 'Y' || linha.charAt(i) != 'Z' || linha.charAt(i) == ';' || linha.charAt(i) == ':' || linha.charAt(i) == 
        	'/' || linha.charAt(i) == '?' || linha.charAt(i) == '!' || linha.charAt(i) == '.' || linha.charAt(i) == ','){
				return resp0;
			}
        }
        return resp1;
	}
	
	public static boolean numero_real(String linha){ //metodo para conferir se a linha possui apenas números reais	
		boolean resp0 = false;
		boolean resp1 = true;
		for(int i = 0; i < linha.length(); i++){	//loop q se repete enquanto i < tamanho da linha			
        	if(linha.charAt(i) != 'a' || linha.charAt(i) != 'e' || linha.charAt(i) != 'i' || linha.charAt(i) != 'o' || linha.charAt(i) != 
        	'u' || linha.charAt(i) != 'A' || linha.charAt(i) != 'E' || linha.charAt(i) != 'I' || linha.charAt(i) != 'O' || 
        	linha.charAt(i) != 'U'|| linha.charAt(i) != 'b' || linha.charAt(i) != 'c' || linha.charAt(i) != 'd' || linha.charAt(i) != 'f' 
        	|| linha.charAt(i) != 'g' || linha.charAt(i) != 'h' ||linha.charAt(i) != 'j' || linha.charAt(i) != 'k' || linha.charAt(i) != 
        	'l' || linha.charAt(i) != 'm' || linha.charAt(i) != 'n' || linha.charAt(i) != 'p' || linha.charAt(i) != 'q' || 
        	linha.charAt(i) != 'r' || linha.charAt(i) != 's' || linha.charAt(i) != 't' || linha.charAt(i) != 'v' || linha.charAt(i) != 
        	'w' || linha.charAt(i) != 'x' || linha.charAt(i) != 'y' || linha.charAt(i) != 'z' || linha.charAt(i) != 'B' || 
        	linha.charAt(i) != 'C' || linha.charAt(i) != 'D' || linha.charAt(i) != 'F' || linha.charAt(i) != 'G' || linha.charAt(i) != 
        	'H' || linha.charAt(i) != 'J' || linha.charAt(i) != 'K' || linha.charAt(i) != 'L' || linha.charAt(i) != 'M' || 
        	linha.charAt(i) != 'N' || linha.charAt(i) != 'P' || linha.charAt(i) != 'Q' || linha.charAt(i) != 'R' || linha.charAt(i) != 
        	'S' || linha.charAt(i) != 'T' || linha.charAt(i) != 'V' || linha.charAt(i) != 'W' || linha.charAt(i) != 'X' || 
        	linha.charAt(i) != 'Y' || linha.charAt(i) != 'Z' || linha.charAt(i) == ';' || linha.charAt(i) == ':' || linha.charAt(i) == 
        	'/' || linha.charAt(i) == '?' || linha.charAt(i) == '!'){
				return resp0;
			}
        }
        return resp1;
	}
}










